stream.go

```diff
@@ -60,8 +60,9 @@ type Stream interface {
 	// Context returns the context for this stream.
 	Context() context.Context
 	// SendMsg is generally called by generated code. On error, SendMsg aborts
```

Comment:
How about this:

```
In the event of a connection or stream error:
- Client: io.EOF is returned and the RPC status can be retrieved by repeatedly calling RecvMsg.
- Server: the error is returned directly

In the event of an error caused by the application (e.g. the message size is over the configured limit), an RPC status error is returned to the user.

On any error, the stream is aborted.  On the server-side, the service handler should cancel any outstanding work and return as soon as possible.
```


Another idea, since the semantics are unfortunately different between client and server, despite having the same interface: let's document all of `Stream`'s behavior in `ClientStream` and `ServerStream` and leave comments in `Stream` indicating that the user should find the documentation there instead.  `Stream` probably isn't directly used anyway, so maybe this would be best?

Reply:
> Another idea, since the semantics are unfortunately different between client and server, despite having the same interface: let's document all of Stream's behavior in ClientStream and ServerStream and leave comments in Stream indicating that the user should find the documentation there instead. Stream probably isn't directly used anyway, so maybe this would be best?

I think that's a great idea. Shall I do so, or would you prefer to? In either case, I'll close this PR.